harness/propertyHelper.js

```diff
@@ -1,13 +1,17 @@
 
 function isConfigurable(obj, name) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, name);
     try {
         delete obj[name];
     } catch (e) {
         if (!(e instanceof TypeError)) {
             $ERROR("Expected TypeError, got " + e);
         }
     }
-    return !Object.prototype.hasOwnProperty.call(obj, name);
+    var result = !Object.prototype.hasOwnProperty.call(obj, name);
+    // Restore property (no-op if not deleted)
+    Object.defineProperty(obj, name, descriptor);
```

Comment:
OK, I guess that could come up if you are using a Proxy that there are other side effects, but for ordinary objects, there should be no side effect to deleting a property and re-adding it the same way.

Anyway, sure, we can leave a side effect in this one, but in that case, could we put something else in propertyHelper.js to test for a non-configurable property and not mutate the object? This would be useful to allow for the multiple observations of the same property on the same object (as in the tests in the following patch). Note that we can still not be sure that it won't be side-effecting in the case of Proxies, however.